Formula/javacc.rb

```diff
@@ -0,0 +1,45 @@
+class Javacc < Formula
+  desc "Java parser generator"
+  homepage "https://javacc.org/"
+  url "https://github.com/javacc/javacc/archive/7.0.4.tar.gz"
+  sha256 "a6a2381dfae5fdfe7849b921c9950af594ff475b69fbc6e8568365c5734cf77c"
+
+  depends_on "ant" => :build
+  depends_on :java
+
+  def install
+    system "ant"
+    libexec.install "target/javacc.jar"
+    doc.install Dir["www/doc/*"]
+    (share/"examples").install Dir["examples/*"]
+    %w[javacc jjdoc jjtree].each do |script|
+      bin.write_java_class_script(libexec/"javacc.jar", script, script)
+    end
+  end
+
+  test do
+    src_file = share/"examples/SimpleExamples/Simple1.jj"
+
+    output_file_stem = testpath/"Simple1"
+
+    system bin/"javacc", src_file
+    assert_predicate output_file_stem.sub_ext(".java"), :exist?
+
+    system bin/"jjtree", src_file
+    assert_predicate output_file_stem.sub_ext(".jj.jj"), :exist?
+
+    system bin/"jjdoc", src_file
+    assert_predicate output_file_stem.sub_ext(".html"), :exist?
+  end
+
+  Pathname.class_eval do
```

Comment:
This function is a light modification of https://github.com/Homebrew/brew/blob/1f924db7f390949e21cf483a6af1e2faf4cc8097/Library/Homebrew/extend/pathname.rb#L372 . It could eventually have its place directly in homebrew sources.